examples/storage_plugin/README.md

```diff
@@ -0,0 +1,204 @@
+Storage plugin example
+======================
+This example shows how to write a user-defined Python storage plugin.
+
+Lets assume that you have an instrument that logs a temperature at certain times.
+The output file from a temperature profile measurement may look as follows:
+
+```
+measurements: 4
+time  temperature
+0     20
+10    70
+20    80
+30    85
+```
+
+The first line provides the number of measurements performed in the experiment, followed by a table of time-temperature pairs for each temperature measurement.
+
+
+Creating an entity
+------------------
+From the manual we know that the time is number of minutes since the start of the measurement and that the temperature is measured in degree Celsius.
+This allows us to make a datamodel in the form of a DLite entity for this output:
+
+```json
+{
+  "uri": "http://onto-ns.com/meta/0.1/TempProfile",
+  "description": "Measured temperature profile.",
+  "dimensions": {
+    "n": "Number temperature measurements."
+  },
+  "properties": {
+    "time": {
+      "type": "int",
+      "shape": ["n"],
+      "unit": "min",
+      "description": "Number of minutes since the start of the experiment."
+    },
+    "temperature": {
+      "type": "int",
+      "shape": ["n"],
+      "unit": "degC",
+      "description": "Measured temperature at the given times."
+    }
+  }
+}
+```
+
+This file will we place in a sub-directory called `entities/`.
+
+
+Creating a storage plugin
+-------------------------
+We will now write a Python storage plugin that can instantiate a `TempProfile` entity from the output data of the instrument.
+We will place this storage plugin in the `plugins/` sub-directory with the following content:
+
+```python
+"""Specific DLite storage plugin for time-temperature profile."""
+from pathlib import Path
+import dlite
+
+
+class tempprofile(dlite.DLiteStorageBase):
+    """DLite storage plugin for a temperature profile."""
+    TempProfile = dlite.Instance.from_location(
+        "json", Path(__file__).resolve().parent.parent /
+        "entities" / "TempProfile.json",
+    )
+
+    def open(self, location, options=None):
+        """Opens `location`."""
+        self.location = location
+
+    def close(self):
+        pass
+
+    def load(self, id=None):
```

Comment:
Please add some more comments to these functions.

Reply:
Good idea. Done

Reply:
Since this is an example, I think we could be even more clear about what is happening. E.g. explaining that on line 91 we are instantiating a TempProfile instance and setting the dimension to n, which is the number of measurements

Reply:
Very good points. I have added the two suggested comments.

So far have we used lower case class names in plugins. Please note that the name of the driver for the plugin is taken from the class name and I guess that lower case driver names are preferable.

Reply:
That makes sense :)